cmd/k8s-operator/sts.go

```diff
@@ -35,11 +39,20 @@ import (
 )
 
 const (
+	// Labels that the operator sets on StatefulSets and Pods. If you add a
+	// new label here, do also add it to tailscaleManagedLabels var to
+	// ensure that it does not get overwritten by ProxyClass configuration.
 	LabelManaged         = "tailscale.com/managed"
 	LabelParentType      = "tailscale.com/parent-resource-type"
 	LabelParentName      = "tailscale.com/parent-resource"
 	LabelParentNamespace = "tailscale.com/parent-resource-ns"
 
+	// LabelProxyClass can be set by users on Connectors, tailscale
+	// Ingresses and Services that define cluster ingress or cluster egress,
+	// to specify that configuration in this ProxyClass should be applied to
+	// resources created for the Connector, Ingress or Service.
+	LabelProxyClass = "tailscale.com/proxy-class"
```

Comment:
It being a label rather than an annotation allows for easier selection of the labelled Services/Ingresses i.e https://github.com/tailscale/tailscale/pull/11074/commits/89120606566adc1cb9992045b3f9ae08e6ba25be#diff-775fe747b40bb9eb4f2a78444d5302811bce24f9ab4126083a0b10d99ac58551R236 (it is not possible to filter on an annotation). Alternatively we could achive the same with indexing Services in cache, but that would be extra work and I think a label is fine